release 0.94.0 (#599)
SpikeInterface/spikeinterface: doc/releases/0.94.0.rst

.. _release0.94.0:

SpikeInterface 0.94.0 release notes
-----------------------------------

15th April 2022

Main change:

>  * Refactor waveformextractor with waveform_tools.
>  * Implement Zarr backend for save()
>  * Read IBL compress files. 
>  * Phase shift (destripe) preprocessor
>  * Test are run partially : faster GH actions
>  * Many improvement in sorting compnents: template matching, select_peak, motion_estimation, motion_correction

Reviewer: ```suggestion
 * Refactor WaveformExtractor with waveform_tools.
 * Implement Zarr backend for save()
 * Read IBL compressed files. 
 * Phase shift (destripe) preprocessor
 * Test are run partially : faster GH actions
 * Many improvement in sorting compnents: template matching, select_peaks, motion_estimation, motion_correction
```

core:

>  * Fix typo in SplitSegmentSorting init
>  * Added from_peaks NumpySorting
>  * Start refactor waveformextractor with waveform_tools.
>  * Support for some systems
>  * Fix bug "too many open file" for the new waveforme extractor. 
>  * Add function to remove empty unit
>  * Remove loose version in favor of packaging 
>  * Add chunk_duration concept for jobs. 
>  * Implement Zarr backend for save() 
>  * Add split recording class and function 
>  * Add Recording.remove_channels() 
>  * Updates to zarr, and save sorting to memory 

Reviewer: ```suggestion
 * Fix typo in SplitSegmentSorting init
 * Added NumpySorting.from_peaks()
 * Refactor WaveformExtractor with waveform_tools.
 * Fix bug "too many open files" for the new waveforme extractor. 
 * Add function to remove empty units
 * Remove loose version in favor of packaging 
 * Add chunk_duration concept for jobs
 * Implement Zarr backend for save() 
 * Add split recording class and function 
 * Add Recording.remove_channels() 
 * Add save sorting to memory 
```

extractors:

>  * Reintroduce spikeglx test files
>  * NeuroscopeSortingExtractor
>  * New neo extractors
>  * Add more neualynx versions for testing 
>  * Modify nwb recorder extractor for loading channel name property as channel_ids when available 
>  * Support rel z in nwb 
>  * Try to read openephys probe 
>  * Fixing "KSLabel" extraction in phykilosortextractor 
>  * Fix Nwb 3d locs, scale preproc, and **plot_kwargs 
>  * Fixes a bug on waveclus sorting extractor 
>  * Read IBL compress files. 

Reviewer: ```suggestion
 * Reintroduce spikeglx test files
 * Add NeuroscopeSortingExtractor
 * New neo extractors
 * Add more neualynx versions for testing 
 * Modify nwb recorder extractor for loading channel name property as channel_ids when available 
 * Support 3d positions (rel_z) in NWB 
 * Read openephys probe from probeinterface
 * Fix "KSLabel" extraction in phykilosortextractor 
 * Bug fix on waveclus sorting extractor 
 * Read IBL compressed files
```

  
sorters:

>  * Refactor run_sorters/run_sorter_by_group to better folders/dtypes 

Reviewer: ```suggestion
 * Refactor run_sorters/run_sorter_by_group to better handle folders and key dtypes 
```

  
toolkit:

 * Phase shift (destripe) preprocessor
 * Fix typo in isi violation
 * Addresses batch size bug #565, and also resolves open file warnings 
 * Add nan to d_prime for single unit case 
 * Handle computing of nearest neighbor metrics for units with few spikes 

  
comparison:

 * Fix multisortingcomparison spiketrains 

widgets:

 * Add presence plot
 * Add with_channel_ids for several widgets that use plot_probe 
 * Add optional distinctipy 

export:

  * Fix mismatch between phy and si unit ids
  

sortingcomponents:

>  * Improve template matching
>  * Start clustering sparse
>  * Improve select_peak()
>  * Improve motion_estimation()
>  * Improve motion_correction()

Reviewer: ```suggestion
 * Improvements to template matching
 * Start sparse clustering
 * Improvements to select_peak()
 * Improvements to motion_estimation()
 * Improvements to motion_correction()
```

 
 
  
doc:
 * Documentation updates for quality_metrics 
 * Try install SI from conda env and skip pip

>  * Doc sorting component + various stuff 

Reviewer: ```suggestion
 * Improve documentation for sorting components
```
